### PR TITLE
Make use of connection pooling in most tests

### PR DESCRIPTION
**Pull request by beikov**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/ConnectionProviderInitiator.java [hunk 1]
```diff
@@ -104,6 +104,10 @@ public ConnectionProvider initiateService(
 			return null;
 		}
 
+		if ( configurationValues.get( AvailableSettings.DATASOURCE ) != null ) {
+			return new DatasourceConnectionProviderImpl();
+		}
+
 		final StrategySelector strategySelector = registry.getService( StrategySelector.class );
 		final Object explicitSetting = configurationValues.get( AvailableSettings.CONNECTION_PROVIDER );
 		if ( explicitSetting != null ) {
```
Comment (clhma) [inline hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/ConnectionProviderInitiator.java:113]: This change breaks the possibility to use a custom connection provider class when a datasource is defined in the project.
This is the case for us. But after upgrading to springboot 3.2, we had to downgrade hibernate verion to 6.2.9 to still use a custom connection provider because it is ignored in 6.2.11.
Could you please explain why this change?
Thanks

Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/ConnectionProviderInitiator.java:113]: This was already fixed via https://hibernate.atlassian.net/browse/HHH-17734

**Comment by sebersole** [comment]
This changes a lot more than "just" tests and needs a Jira imo
